### PR TITLE
python2Packages.cfn-lint: fix importlib-resources version bounds

### PR DESCRIPTION
**Pull request by freezeboy**
###### Motivation for this change

package dependencies don't support python 2
ZHF: #97479

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drewrisinger** [comment]
Why is this package being disabled? What's failing? From looking at upstream, it supports ``py27``: https://github.com/aws-cloudformation/cfn-python-lint/blob/027091757862c3d88cb02a264200cfeefdb3a156/setup.py#L59

**Comment by freezeboy** [comment]
importlib-resource is required to be 1.4, and we have 3.0.0 in the repository

**Comment by drewrisinger** [comment]
Applying this patch seemed to work (tested against GitHub source, not PyPi). Don't need to disable python2.7 then.

```
  postPatch = ''
    substituteInPlace setup.py --replace 'importlib_resources~=1.4;python_version<"3.7" and python_version!="3.4"' 'importlib_resources;python_version<"3.7"'
  '';
```

**Comment by freezeboy** [comment]
Thank you for the patch

File: pkgs/development/python-modules/cfn-lint/default.nix [hunk 1]
```diff
@@ -36,7 +40,9 @@ buildPythonPackage rec {
     setuptools
     junit-xml
     networkx
-  ] ++ lib.optionals (pythonOlder "3.8") [ importlib-metadata importlib-resources ];
+  ]
+  ++ #lib.optionals (pythonOlder "3.8")
```
Comment (drewrisinger) [inline pkgs/development/python-modules/cfn-lint/default.nix:44]: Why disable the optionals?

**Comment by drewrisinger** [comment]
Commit should be renamed to something like ``pythonPackages.cfn-lint: ...``. Has a large rebuild via ``nix-review``, so I'm not going to build all locally, but tested working with some packages:

``nix-review pr 97762 -p pythonPackages.cfn-lint -p apache-airflow``
```
https://github.com/NixOS/nixpkgs/pull/97762
2 packages built:
apache-airflow python3Packages.cfn-lint
```

**Comment by risicle** [comment]
Seeing as we're now patching the source, would you be able to add a `pythonImportsCheck =` seeing as we don't have tests enabled here? It's better than nothing.

**Comment by freezeboy** [comment]
As it is supposed to be an app more than a library, I only took the packages in the root folder's `__init__py` file.

Built for python2 and 3 locally

**Review by jonringer — APPROVED** [review]
does in fact fix the build, other failures can be dealt with in other PRs

**Pull request merged**